pkg/controllers/management/auth/role_template_handler.go

```diff
@@ -1,47 +1,105 @@
 package auth
 
 import (
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/clustermanager"
 	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	"github.com/rancher/types/config"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/cache"
 )
 
 const (
-	roleTemplateHandlerName = "mgmt-auth-role-template-handler"
+	roleTemplateHandlerName = "mgmt-auth-roletemplate-handler"
 	prtbByRoleTemplateIndex = "management.cattle.io/prtb-by-role-template"
 )
 
 type roleTemplateHandler struct {
-	prtbIndexer cache.Indexer
-	prtbClient  v3.ProjectRoleTemplateBindingInterface
+	prtbIndexer    cache.Indexer
+	prtbClient     v3.ProjectRoleTemplateBindingInterface
+	clusters       v3.ClusterInterface
+	clusterManager *clustermanager.Manager
 }
 
-func newRoleTemplateHandler(management *config.ManagementContext) *roleTemplateHandler {
+func newRoleTemplateHandler(management *config.ManagementContext, clusterManager *clustermanager.Manager) v3.RoleTemplateLifecycle {
```

Comment:
Since this is now a lifecycle, can we name the function accordingly?

Reply:
file name as well

Reply:
and the name `roleTemplateHandler` the functions are attached to

Reply:
Renamed everything